### PR TITLE
Hermetic tests for call screen for different roles

### PR DESCRIPTION
**Pull request by prabhjot-msft**
# What
Hermetic tests for call screen for different roles

# Why
https://skype.visualstudio.com/SPOOL/_workitems/edit/2945144

# How Tested

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/react-composites/tests/browser/call/hermetic/RoomsCallScreen.test.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation.
```
Comment (JamesBurnside) [inline packages/react-composites/tests/browser/call/hermetic/RoomsCallScreen.test.ts:1]: Question looking at the snapshots - are we (or are we planning to) update the tooltips to say why certain controls are disabled?
image

e.g. say something like `You do not have permission to start your camera in this call`

Comment (JamesBurnside) [reply packages/react-composites/tests/browser/call/hermetic/RoomsCallScreen.test.ts:1]: If we do, we should add snapshot tests that capture the tooltip

File: packages/react-composites/tests/browser/call/hermetic/RoomsCallScreen.test.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+// Licensed under the MIT license.
+
+import { buildUrlWithMockAdapter, defaultMockCallAdapterState, test } from './fixture';
+import { expect } from '@playwright/test';
+import { dataUiId, stableScreenshot, waitForSelector } from '../../common/utils';
+import { IDS } from '../../common/constants';
+
+test.describe('Rooms CallScreen tests for different roles', async () => {
+  test('All CallControls are enabled for Presenter', async ({ page, serverUrl }) => {
+    const initialState = defaultMockCallAdapterState();
```
Comment (dmceachernmsft) [inline packages/react-composites/tests/browser/call/hermetic/RoomsCallScreen.test.ts:11]: should we be adding any remote participants in these tests? what does this look like if you are a consumer and one of the remote participants is the organizer for example?

File: packages/react-composites/tests/browser/call/hermetic/RoomsCallScreen.test.ts (hunk 1 above)
Comment (prabhjot-msft) [reply packages/react-composites/tests/browser/call/hermetic/RoomsCallScreen.test.ts:1]: We had the discussion around that and we are not going to have specific tooltips for rooms as of now. If we add the tooltips in future, will add the tests for same :)

File: packages/react-composites/tests/browser/call/hermetic/RoomsCallScreen.test.ts (hunk 2 above)
Comment (prabhjot-msft) [reply packages/react-composites/tests/browser/call/hermetic/RoomsCallScreen.test.ts:11]: I have added the remote participants for the tests for consumer, but we cannot control the role for remote participant as it is passed as a prop to call composite as of now. Let me know if you have any thoughts on that.

**Review by dmceachernmsft — APPROVED** [review]

File: packages/react-composites/tests/browser/call/hermetic/RoomsCallScreen.test.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    await page.goto(buildUrlWithMockAdapter(serverUrl, initialState, { role: 'Presenter' }));
+    await waitForSelector(page, dataUiId(IDS.videoGallery));
+    expect(await stableScreenshot(page, { dismissTooltips: true })).toMatchSnapshot('rooms-call-screen-presenter.png');
+  });
+
+  test('All CallControls are enabled for Attendee', async ({ page, serverUrl }) => {
```
Comment (JamesBurnside) [inline packages/react-composites/tests/browser/call/hermetic/RoomsCallScreen.test.ts:17]: Screenshare should not be enabled for an attendee (correct in the snapshots but this test text is incorrect)

File: packages/react-composites/tests/browser/call/hermetic/RoomsCallScreen.test.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    await page.goto(buildUrlWithMockAdapter(serverUrl, initialState, { role: 'Attendee' }));
+    await waitForSelector(page, dataUiId(IDS.videoGallery));
+    expect(await stableScreenshot(page, { dismissTooltips: true })).toMatchSnapshot('rooms-call-screen-attendee.png');
+  });
+
+  test('Only few CallControls are enabled for Consumer', async ({ page, serverUrl }) => {
+    const initialState = defaultMockCallAdapterState();
+    await page.goto(buildUrlWithMockAdapter(serverUrl, initialState, { role: 'Consumer' }));
+    await waitForSelector(page, dataUiId(IDS.videoGallery));
+    expect(await stableScreenshot(page, { dismissTooltips: true })).toMatchSnapshot('rooms-call-screen-consumer.png');
```
Comment (JamesBurnside) [inline packages/react-composites/tests/browser/call/hermetic/RoomsCallScreen.test.ts:28]: I know this is just a hermetic fake-data artifact, but is odd that the participant is stuck in the unmuted state:
image

File: packages/react-composites/tests/browser/call/hermetic/RoomsCallScreen.test.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  });
+
+  test('Only few CallControls are enabled for Consumer with remote participants', async ({ page, serverUrl }) => {
```
Comment (JamesBurnside) [inline packages/react-composites/tests/browser/call/hermetic/RoomsCallScreen.test.ts:31]: Is there value to having two tests, one with remote participants and one without?

File: packages/react-composites/tests/browser/call/hermetic/RoomsCallScreen.test.ts (hunk 4 above)
Comment (JamesBurnside) [reply packages/react-composites/tests/browser/call/hermetic/RoomsCallScreen.test.ts:28]: (I don't think we can do anything about this here, the call state should just always be muted when in this role)

**Review by JamesBurnside — APPROVED** [review]

File: packages/react-composites/tests/browser/call/hermetic/RoomsCallScreen.test.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (prabhjot-msft) [reply packages/react-composites/tests/browser/call/hermetic/RoomsCallScreen.test.ts:17]: Yeah, I am writing tests for that. There will be tests covering remove button as well checking device lists.

File: packages/react-composites/tests/browser/call/hermetic/RoomsCallScreen.test.ts (hunk 4 above)
Comment (prabhjot-msft) [reply packages/react-composites/tests/browser/call/hermetic/RoomsCallScreen.test.ts:28]: Yeah, we are using same default state in all mock tests and user is unmuted by default in mock state, that is not the case in actual call though :)

File: packages/react-composites/tests/browser/call/hermetic/RoomsCallScreen.test.ts (hunk 5 above)
Comment (prabhjot-msft) [reply packages/react-composites/tests/browser/call/hermetic/RoomsCallScreen.test.ts:31]: Not really, added this test later will only keep the one with remote participants, thanks for noticing

**Review by carocao-msft — APPROVED** [review]

**Pull request merged**